ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -189,72 +262,304 @@ def waypoints_cb(self, lane_msg):
+        # this generates the deceleration curve based on JMT
+        # for the range of waypoints, inserting V = 0.0
+        # beyond the stop point
+        recalc = False
+        curpt = self.waypoints[start_ptr]
+        target_velocity = 0.0
+
+        if curpt.JMT_ptr == -1 or self.state != 'slowdown':
+            jmt_ptr = self.setup_jmt(curpt, target_velocity, safety_factor,
+                                     self.dyn_jmt_time_factor)
+            curpt.JMT_ptr = jmt_ptr
+            self.state = 'slowdown'
+        else:
+            jmt_ptr = curpt.JMT_ptr
+            rospy.loginfo("using old jmt_ptr = {}".format(jmt_ptr))
+        JMT_instance = self.JMT_List[jmt_ptr]
+
+        t = 0.0
+        for ptr in range(start_ptr + 1, start_ptr + num_wps):
+            mod_ptr = ptr % len(self.waypoints)
+            curpt = self.waypoints[mod_ptr]
+
+            if curpt.get_s() <= JMT_instance.final_displacement:
+                # create the main part of the jmt curve
+                t = self.gen_point_in_jmt_curve(curpt, jmt_ptr, t)
+                if self.check_point(curpt) is True:
+                    recalc = True
+            else:
+                # the car has reached the point where the JMT
+                # curve should reach target velocity of 0.0
+                # we smooth the transition
+                rospy.loginfo("{} beyond S = {} at ptr_id = {}".
+                              format(curpt.get_s(), JMT_instance.
+                                     final_displacement, mod_ptr))
+                self.set_transition_to_stop(mod_ptr)
+        return recalc
+
+    def maintain_speed(self, start_ptr, num_wps):
+        # sets the speed of the car to default velocity for the
+        # range of waypoints
+        for ptr in range(start_ptr, start_ptr + num_wps):
+            mod_ptr = ptr % len(self.waypoints)
+            curpt = self.waypoints[mod_ptr]
+
+            rospy.loginfo("setting to default velocity {} at ptr = {}"
+                          .format(self.default_velocity, mod_ptr))
+
+            velocity = min(curpt.get_maxV(), self.default_velocity)
+            curpt.set_v(velocity)
+            curpt.JMTD.set_VAJt(velocity, 0.0, 0.0, 0.0)
+            curpt.JMT_ptr = -1
+
+    def init_acceleration(self, start_ptr, num_wps):
+        # starts the car moving with a lead foot
+        offset = 0
+
+        velocity = max(0.4, self.waypoints[start_ptr].get_v())
+        self.waypoints[start_ptr].set_v(velocity)
+        self.waypoints[start_ptr].JMTD.set_VAJt(
+                    velocity, self.default_accel, 0.0, 0.0)
+        offset += 1
+        while velocity < 1.5 and offset < num_wps:
+            disp = self.waypoints[start_ptr + offset].get_s() -\
+                self.waypoints[start_ptr].get_s()
+            velocity = min(max(0.5, math.sqrt(self.default_accel * disp * 2.0)),
+                           self.waypoints[start_ptr + offset].get_maxV())
```

Comment:
The "0.4" on line 391, "1.5" on line 396, and "0.5" on line 399 should probably be set as named parameter values for tuning (doesn't need to be dynamic reconfigure, just local parameters).  I think the 1.5 value should be reduced for smoother initial tip in acceleration, maybe 1.0 or 0.5.  For the 0.4/0.5 velocity values instead of trying to pick some initial small arbitrary velocity target to start the car rolling, maybe it's better to initially set waypoint offset 0 velocity as 0, then calculate offset 1 velocity from the tuned accel rate, and then set the velocity at offset 0 to be equal to some ratio of the velocity at offset 1 (like half or even just set equal).  This area is tricky and will need more tweaking as all the pieces are closer to final.

Reply:
Will switch to named values